### PR TITLE
Feature/issue 7

### PR DESCRIPTION
**Pull request by bbangert**
Refers to #7. This is the aggregation of other important push-service relevant documentation. Some of the bits not as related to the server are not being pulled over.

@jrconlin r?

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+**Example**
+
+     {
+       "messageType": "register",
+       "channelID": "d9b74644-4f97-46aa-b8fa-9393985cd6cd",
+       "status": 200,
+       "pushEndpoint": "http://pushserver.example.org/d9b74644"
+     }
+
```
Comment (jrconlin) [inline docs/design.md:189]: Bridge functions like GCM may include additional bits of data, thus probably not a good idea to create a strict YAML for this. If that's a problem we could put the data in an "extra" block that can be ignored by strict interpreters (e.g. Go, Java, etc). Thoughts?

File: docs/index.md [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+with the [SimplePush DOM
+API](https://developer.mozilla.org/en-US/docs/Web/API/Simple_Push_API) that runs
+under [Firefox OS][fxos]. This API only supported incrementing version numbers,
+to wake a device so that it could check with an Application Server to determine
+what to display.
+
+In 2015 the Push Service was rewritten in Python, and added preliminary
+[WebPush][wp] support for carrying data in push messages. The underlying
+protocol used between [Firefox][ffx] and the Push Service is an extended
+[SimplePush Protocol][] utilizing a websocket protocol that will eventually be
```
Comment (jrconlin) [inline docs/index.md]: Link?

File: docs/index.md [hunk 3]
```diff
@@ -23,13 +40,97 @@ Additional ways to get in contact with the team:
...
+: A service that delivers push messages to user agents.
+
+SimplePush
+: Firefox OS specific Push system that carries no data, only an incrementing
+  version number. The Mozilla Push Service continues to support this legacy API
+  for Firefox OS devices.
+
+UAID
+: A globally unique UserAgent ID. Used by the Push Service to associate clients
+  utilizing the websocket protocol with their associated channel ID's.
```
Comment (jrconlin) [inline docs/index.md]: Nit: Should we note that this is a UUID4 here or suppose that the developer will later read that?

**Comment by jrconlin** [comment]
1 Q, 1 R and 1 Nit. 
Otherwise LGTM
r + Pending link

File: docs/design.md (hunk 1 above)
Comment (bbangert) [inline docs/design.md:189]: I didn't include GCM here because I thought we were doing that 100% via the RESTful API now.

**Comment by bbangert** [comment]
Fixed the link, and added the UAID bit about it being a UUID.

File: docs/design.md (hunk 1 above)
Comment (jrconlin) [inline docs/design.md:189]: And that explains why the REST portions are not yet included. Fair enough. Nevermind.

**Comment by jrconlin** [comment]
r+

File: docs/design.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+channelID string REQUIRED
+: ChannelID of the notification
+
+version string REQUIRED
+: Version of the notification sent
+
+headers map OPTIONAL
+: Encryption headers sent along with the data. Present only if data was sent.
+
+data string OPTIONAL
```
Comment (ghost) [inline docs/design.md:344]: It might be worth noting that this is Base64 URL-encoded.

Comment (bbangert) [inline docs/design.md:344]: Indeed, was already merged though, so I'll file this for another bug. If you see anything else add it to Issue #10

File: docs/design.md [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+#### UserAgent -> PushServer
+
+messageType = "hello"
+: Begin handshake
+
+uaid string REQUIRED
+: If the UserAgent has a previously assigned UAID, it should send it. Otherwise
+  send an empty string.
+
+channelIDs list of strings REQUIRED
```
Comment (ghost) [inline docs/design.md:95]: Only required for Simple Push. The client no longer sends these for Web Push.

**Pull request merged**